collector/interface.go

```diff
@@ -214,7 +222,23 @@ func processIfaceNetconfReply(reply *netconf.RPCReply, ch chan<- prometheus.Metr
 		return fmt.Errorf("could not unmarshal netconf reply xml: %s", err)
 	}
 	for _, ifaceData := range netconfReply.InterfaceInformation.PhysicalInterface {
+		ifaceDesc := getInterfaceDesc()
 		ifaceLabels := []string{strings.TrimSpace(ifaceData.Name.Text)}
+
+		var allIfaceKeys map[string]interface{}
```

Comment:
strategy here was:
- create an interface to unmartial the json field in the interface description into
- iterate through the configured interface description keys to add to the metric and either add the value retrieved from the interface by the key, or add an empty string

note this is only for the physical interfaces; we're not adding optional keys to logical interfaces (should maybe make this a config file flag?)